homeassistant/components/mqtt/client.py

```diff
@@ -404,12 +410,17 @@ def __init__(
         self._ha_started = asyncio.Event()
         self._cleanup_on_unload: list[Callable[[], None]] = []
 
-        self._paho_lock = asyncio.Lock()  # Prevents parallel calls to the MQTT client
+        self._connection_lock = asyncio.Lock()
```

Comment:
I did keep a lock on the connection because it is possible to shutdown/unload the config entry while its still connecting